src/kernels/jupyter/jupyterKernelService.node.ts

```diff
@@ -276,75 +271,38 @@ export class JupyterKernelService implements IJupyterKernelService {
                     );
                     specModel.argv[0] = interpreter.uri.fsPath;
                 }
-                // Get the activated environment variables (as a work around for `conda run` and similar).
-                // This ensures the code runs within the context of an activated environment.
-                const interpreterEnv = await this.activationHelper
-                    .getActivatedEnvironmentVariables(resource, interpreter, true)
-                    .catch(noop)
-                    // eslint-disable-next-line @typescript-eslint/no-explicit-any
-                    .then((env) => (env || {}) as any);
-
-                // Ensure we inherit env variables from the original kernelspec file.
-                let envInKernelSpecJson =
-                    getKernelRegistrationInfo(kernel) === 'registeredByNewVersionOfExtForCustomKernelSpec'
-                        ? specModel.env || {}
-                        : {};
-
-                // Give preferences to variables in the env file (except for `PATH`).
-                envInKernelSpecJson = Object.assign({ ...interpreterEnv }, envInKernelSpecJson);
-                if (interpreterEnv['PATH']) {
-                    envInKernelSpecJson['PATH'] = interpreterEnv['PATH'];
-                }
-                if (interpreterEnv['Path']) {
-                    envInKernelSpecJson['Path'] = interpreterEnv['Path'];
-                }
-                specModel.env = Object.assign(envInKernelSpecJson, specModel.env);
-
-                // Ensure the python env folder is always at the top of the PATH, this way all executables from that env are used.
-                // This way shell commands such as `!pip`, `!python` end up pointing to the right executables.
-                // Also applies to `!java` where java could be an executable in the conda bin directory.
-                if (specModel.env) {
-                    this.envVarsService.prependPath(specModel.env as {}, path.dirname(interpreter.uri.fsPath));
-                }
-
-                // If user asks us to, set PYTHONNOUSERSITE
-                // For more details see here https://github.com/microsoft/vscode-jupyter/issues/8553#issuecomment-997144591
-                // https://docs.python.org/3/library/site.html#site.ENABLE_USER_SITE
```

Comment:
This duplicate code can be replaced with `KernelEnvironmentVariablesService.getEnvironmentVariables`

Reply:
Now both raw kernels and jupyter kernels will use the same set of code to generate env variables for staring of kernels.